### PR TITLE
Vote extension propagation (RFC017)

### LINKED ISSUE #9852: Vote extension propagation
Cherry-pick #8402

Handling of propagation of vote extensions according to RFC017

### PR DESCRIPTION
**Pull request by sergio-mena**
Closes #9852

The first commit is a cherry-pick of #8433.
Subsequent commits are fixing the build, and the unit tests

N.B: The changelog will be updated once branch `feature/abci++vef` is merged back into `main`

---

#### PR checklist

- [x] Tests written/updated, or no tests needed
- [x] `CHANGELOG_PENDING.md` updated, or no changelog entry needed
- [x] Updated relevant documentation (`docs/`) and code comments, or no
      documentation updates needed

### CODE REVIEW

File: blocksync/reactor.go [hunk 1]
```diff
@@ -334,10 +365,14 @@ FOR_LOOP:
 			// routine.
 
 			// See if there are any blocks to sync.
-			first, second := bcR.pool.PeekTwoBlocks()
+			first, second, extCommit := bcR.pool.PeekTwoBlocks()
 			// bcR.Logger.Info("TrySync peeked", "first", first, "second", second)
```
Comment (williambanfield) [inline blocksync/reactor.go:369]: Can we remove this commented out code while we're here?

File: consensus/state_test.go [hunk 2]
```diff
@@ -1668,10 +1668,12 @@ func TestFinalizeBlockCalled(t *testing.T) {
 				Status: abci.ResponseProcessProposal_ACCEPT,
 			}, nil)
 			m.On("PrepareProposal", mock.Anything, mock.Anything).Return(&abci.ResponsePrepareProposal{}, nil)
-			m.On("ExtendVote", mock.Anything, mock.Anything).Return(&abci.ResponseExtendVote{}, nil)
-			m.On("VerifyVoteExtension", mock.Anything, mock.Anything).Return(&abci.ResponseVerifyVoteExtension{
-				Status: abci.ResponseVerifyVoteExtension_ACCEPT,
-			}, nil)
+			if !testCase.voteNil {
```
Comment (williambanfield) [inline consensus/state_test.go:1671]: Looks like there's a comment missing from the hunk here.

**Review by williambanfield — APPROVED** [review]
Did a pass looking at the diff-of-diffs. Looks like a clean cherry-pick.

File: blocksync/reactor.go (hunk 1 above)
Comment (sergio-mena) [reply blocksync/reactor.go:369]: This is done in the upcoming cherry-pick. Bear with me

File: consensus/state_test.go (hunk 2 above)
Comment (sergio-mena) [reply consensus/state_test.go:1671]: Thanks for catching this! Adding it

**Pull request merged**